ocaml-lsp-server/src/dune.mli

```diff
@@ -0,0 +1,12 @@
+open Import
+
+type state =
+  | Absent
+  | Out_of_date
+  | Running
+
+type t
+
+val state : t -> state Fiber.t
+
+val create : Scheduler.t -> t
```

Comment:
3. Isn't it quite unexpected that the function `state` can start a process and mutate things? Yes, we return a fiber, but it still seems suspicious. Wdyt?
4. (minor) While use of `create` would signal that some mutation (launching a process) is happening (vs `make`, for example).

Reply:
3. It cannot mutate things in a way that is observable to users of the module.
4. That's how I did it initially but it came with its own problems. Like now we need to wait for `create` to complete before we can finish initializing the server.